packages/babel-parser/src/parser/util.js

```diff
@@ -3,7 +3,9 @@
 import { types as tt, type TokenType } from "../tokenizer/types";
 import Tokenizer from "../tokenizer";
 import type { Node } from "../types";
-import { lineBreak } from "../util/whitespace";
+import { lineBreak, skipWhiteSpace } from "../util/whitespace";
+
+const literal = /^(?:'((?:\\.|[^'])*?)'|"((?:\\.|[^"])*?)"|;)/;
```

Comment:
I took 5 mins to understand what is this lol

I think that it could be simplified a bit:
```js
/^(?:;|('|")((?:\\?.)*?)\1)/
```

1. You don't need the `[^"]` part, because you are already using `*?` so it stops as soon as possible (and the closing quote this is parsed as the end of the literal).
2. You can merge the `"` and `'` cases, using `\1` to refer to the matched quotes kind.
3. Since this adds a new capturing group, you would have to chech `match[2]`.

Demo: https://regex101.com/r/aQTDzs/1

Reply:
Thanks, that is way easier to read.